### DISCUSSION TITLE
Can't get caddy to work with AWS EC2: "could not get certificate from issuer"

### DISCUSSION THREAD

**Discussion by enpro-github** [Budibase/budibase#4536] [Help]
I've been following the docs on how to use caddy to setup https with our instance. But it's not as simple as it shows for me. When I run 
`caddy reverse-proxy --from yourdomain.com --to localhost:10000` I see a bunch of warnings and errors: 
![image](https://user-images.githubusercontent.com/68210951/154373000-1a130ad5-8b8f-4c87-ad92-fdd22839e8a6.png)

I have ports 80 and 443 open on my ec2 as well

**Comment by shogunpurple** [comment]
@enpro-github have you run a command like this?

`caddy reverse-proxy --from <your-ec2-hostname> --to localhost:10000` ?

This is not how HTTPS on a domain works - you need to actually buy a domain, and point that domain to your EC2 instance. Then you run:

`caddy reverse-proxy --from <your-domain> --to localhost:10000` and caddy will obtain and verify the SSL certificate for you.

**Reply by enpro-github** [reply]
Thanks for the speedy reply, that was actually the first method I tried, shouldve included that. I assume also I sholdnt include the subdomain right? It's saying It can't get the certificate from issuer. Should I be making sure there's a cert installed somewhere?

**Reply by shogunpurple** [reply]
Caddy should do this for you and perform the DNS challenge. No, you don't include the subdomain when setting up the reverse proxy for your top level domain. Can you confirm that you have:

- Bought a domain, and if so, where from
- Configured an A record in your DNS settings that points to the IP address of your EC2 instance
- Are able to access your EC2 instance on port 80 and 443
- SSHed into EC2 instance ran `caddy reverse-proxy --from <your-top-level-domain> --to localhost:10000`

If you want subdomains:
- Set up a CNAME record to point your subdomain to your top level domain
- Update your `Caddyfile` to look like this:
```
example.com {
    reverse_proxy localhost:10000
}

subdomain2.example.com {
    reverse_proxy localhost:10000
}
```
- run `caddy run`

**Reply by enpro-github** [reply]
We bought the domain from Netfirms then transferred it to GoDaddy in 2021.  We also configured the A record to point to our IP address: 
![image](https://user-images.githubusercontent.com/68210951/154410000-30b42183-fc84-4e75-ae23-47ad2fde5a88.png)

My EC2 instance is configured to allow inbound traffic on ports 80 and 443:
![image](https://user-images.githubusercontent.com/68210951/154410203-e576b8e2-32d5-493b-86fa-e61e4702d8bf.png)
![image](https://user-images.githubusercontent.com/68210951/154410264-883bf478-62e4-4b1a-88c5-d78a371fa262.png)
I have SSHed into the instance and ran that command

**Reply by enpro-github** [reply]
What if our subdomain is hosted at a different IP than our domain? Our subdomain currently already has an A record pointing the the IP of the host of our domain

**Reply by enpro-github** [reply]
Hey I've made some more leeway, I put this project down and came back to it. Am I supposed to create an entry for Caddy in my docker file?
